### PR TITLE
Add Bucket Access Logging. Authors: @claudiodavid-ai @KishanAnjanaKha…

### PR DESCRIPTION
**Pull request by Claudiodavid-ai**
We added access logging option for the encrypted bucket.
We have added a test assertions that's currently failing. We think it's a bug from the awspec library. Have you encounters this type of issue with the library before? I see there are 2 other asserts that are failing for mfa_delete and server_side_encryption.

How can we move forward with this change?

### CODE REVIEW

**Comment by tobyclemson** [comment]
Hi @Claudiodavid-ai,

I'll take a look this weekend. Often when awspec doesn't work for me I drop down to the AWS client directly.

Thanks,
Toby

**Comment by Gryff** [comment]
Hi @Claudiodavid-ai 
I've fixed the pipeline on the main branch, unfortunately this has caused merge conflicts in your PR.
If you're still interested in getting this PR merged, please resolve the conflicts and then I can take a look.
If not, no worries, I'll close this after a few weeks.

**Comment by Claudiodavid-ai** [comment]
Thanks. I'll try to find some time in the next two days to fix it.

**Comment by Claudiodavid-ai** [comment]
Done. No more merge conflicts.
I didn't have time to run the tests yet though 😄 .

File: README.md [hunk 1]
```diff
@@ -31,6 +31,8 @@ for more details.
 | source_policy_json                          | A source policy for the bucket, additional statements to enable encryption will be added to the policy              |      ""      |    no    |
 | acl                                         | The [canned ACL](https://docs.aws.amazon.com/AmazonS3/latest/dev/acl-overview.html#canned-acl) to apply.            |   private    |    no    |
 | mfa_delete                                  | Enable MFA delete for either _Change the versioning state of your bucket_ or _Permanently delete an object version_ |    false     |    no    |
+| enable_access_logging                       | Enable access logging for the bucket. Also creates the logging bucket                                               | no           | no       |
+| bucket_key_enabled                          | Whether or not to use [S3 Bucket Keys](https://docs.aws.amazon.com/AmazonS3/latest/userguide/bucket-key.html)       | false        | no       |
```
Comment (Gryff) [inline README.md:35]: `access_log_bucket_name` needs to be added here too

**Comment by Gryff** [comment]
Can you also add the new feature to the changelog? It will be released in v2.1.0

File: README.md (hunk 1 above)
Comment (Claudiodavid-ai) [reply README.md:36]: Done.

**Comment by tobyclemson** [comment]
Hey @Claudiodavid-ai,

I think it would be simpler if rather than this module trying to create the access logging bucket itself, it accepted a bucket ARN for the access log bucket and used that instead. That way, how the access log bucket gets created (i.e., whether it uses encryption or not) is up to the consumer. The consumer could even use this module to create the access log bucket to then pass in to the module instance to create the bucket to be logged, if you see what I mean.

I'm happy to merge this in now and make some changes on our end but wanted you to be aware, the interface wouldn't be the same once released.

Thanks,
Toby

**Comment by Claudiodavid-ai** [comment]
👍  Makes sense. I think it would be great for the `target_prefix` under `logging` section to be configurable to allow you to use the same bucket with different paths as the logging target for multiple buckets.

**Comment by tobyclemson** [comment]
Good idea, I'll incorporate that into the refactor.

**Comment by tobyclemson** [comment]
Good idea, I'll incorporate that before we release.

**Comment by tobyclemson** [comment]
Good idea, I'll incorporate that before we release.

**Comment by tobyclemson** [comment]
@Claudiodavid-ai this is available in [2.2.0](https://registry.terraform.io/modules/infrablocks/encrypted-bucket/aws/2.2.0)

**Pull request merged**